### PR TITLE
Link against libm for Unix (issue#213).

### PR DESCRIPTION
**Pull request by badshah400**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,6 +99,9 @@ if(PNG_FOUND)
   set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} ${PNG_LIBRARIES})
 endif(PNG_FOUND)
 
+if(UNIX)
+  link_libraries(m)
```
Comment (mathstuf) [inline CMakeLists.txt:103]:
```suggestion
  list(APPEND ADDITIONAL_LIBRARIES m)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -99,6 +99,9 @@ if(PNG_FOUND)
   set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} ${PNG_LIBRARIES})
 endif(PNG_FOUND)
 
+if(UNIX)
```
Comment (madebr) [inline CMakeLists.txt:102]: - [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html)
- [`UNIX`](https://cmake.org/cmake/help/latest/variable/UNIX.html) is also true on Macos
```suggestion
if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Comment (mathstuf) [reply CMakeLists.txt:102]: Don't you want/need this on macOS as well? Certainly things like FreeBSD and friends should have it too. Maybe `UNIX AND NOT APPLE` perhaps?

Comment (madebr) [reply CMakeLists.txt:102]: I don't own any Apple things myself, but I remember somebody saying that linking to the `m` library is not needed there.
See this stackoverflow: https://stackoverflow.com/questions/33675638/gcc-link-the-math-library-by-default-in-c-on-mac-os-x
Your suggestion looks good.

Comment (badshah400) [reply CMakeLists.txt:102]: So, would you still prefer I modified `if(UNIX)` to `if(UNIX AND NOT APPLE)` or is the PR ok as it is?

Comment (madebr) [reply CMakeLists.txt:102]: I don't know about the BSD's, I know the `m` library is ok on Macos (but not needed).
So I think it's ok.
There is no ci to confirm it though.

**Comment by mathstuf** [comment]
Any progress here?

**Comment by badshah400** [comment]
Sorry, do I need to do something more here?

**Comment by mathstuf** [comment]
I don't know. The open discussion is about the platforms which need this. Does anyone have a BSD and/or macOS install handy to test those platform requirements?

**Comment by bramton** [comment]
Hopefully fixed: https://github.com/libharu/libharu/commit/df5a9c364d53d9be1e746a9a86c612bbd0451b4c

**Pull request closed without merging**